helpers.js

```diff
@@ -0,0 +1,20 @@
+const { Client: ScpClient } = require("node-scp");
+const fs = require("fs/promises");
+
+async function getScpClient(params) {
+  const privateKey = params.privateKey || (
+    params.keyPath && await fs.readFile(params.keyPath)
+  );
+
+  return ScpClient({
+    host: params.host,
+    port: params.port ?? 22,
```

Comment:
Since you're refactoring `config.json`, you could add this as the default value.